For gcc 4.7 unistd.h needs to be included.
See:
https://bugzilla.mozilla.org/show_bug.cgi?id=736765
Also fixed in FF 12
diff -Naur mozilla-release.orig/ipc/chromium/src/base/file_util.cc mozilla-release/ipc/chromium/src/base/file_util.cc
--- mozilla-release.orig/ipc/chromium/src/base/file_util.cc	2012-03-13 01:36:29.000000000 +0000
+++ mozilla-release/ipc/chromium/src/base/file_util.cc	2012-03-18 01:23:05.750834540 +0000
@@ -8,7 +8,7 @@
 #include <io.h>
 #endif
 #include <stdio.h>
-#if defined(ANDROID)
+#if defined(ANDROID) || defined(OS_POSIX)
 #include <unistd.h>
 #endif
 
diff -Naur mozilla-release.orig/ipc/chromium/src/base/file_util_linux.cc mozilla-release/ipc/chromium/src/base/file_util_linux.cc
--- mozilla-release.orig/ipc/chromium/src/base/file_util_linux.cc	2012-03-13 01:36:29.000000000 +0000
+++ mozilla-release/ipc/chromium/src/base/file_util_linux.cc	2012-03-18 01:23:05.757501140 +0000
@@ -5,6 +5,7 @@
 #include "base/file_util.h"
 
 #include <fcntl.h>
+#include <unistd.h>
 
 #include <string>
 #include <vector>
diff -Naur mozilla-release.orig/ipc/chromium/src/base/message_pump_libevent.cc mozilla-release/ipc/chromium/src/base/message_pump_libevent.cc
--- mozilla-release.orig/ipc/chromium/src/base/message_pump_libevent.cc	2012-03-13 01:36:29.000000000 +0000
+++ mozilla-release/ipc/chromium/src/base/message_pump_libevent.cc	2012-03-18 01:23:05.760834441 +0000
@@ -6,6 +6,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #include "eintr_wrapper.h"
 #include "base/logging.h"
diff -Naur mozilla-release.orig/ipc/chromium/src/base/time_posix.cc mozilla-release/ipc/chromium/src/base/time_posix.cc
--- mozilla-release.orig/ipc/chromium/src/base/time_posix.cc	2012-03-13 01:36:29.000000000 +0000
+++ mozilla-release/ipc/chromium/src/base/time_posix.cc	2012-03-18 01:23:05.767501042 +0000
@@ -8,6 +8,7 @@
 #include <mach/mach_time.h>
 #endif
 #include <sys/time.h>
+#include <unistd.h>
 #ifdef ANDROID
 #include <time64.h>
 #else
